### ISSUE TITLE
Plotting: adding points for the vertices by default or not?

### ISSUE THREAD

**Issue by jorisvandenbossche** [shapely/shapely#1550]
https://github.com/shapely/shapely/pull/1497 added an experimental plotting module to shapely. There was one remaining discussion point that was not yet resolved: should the function to plot polygons and lines by default plot points as well, and is a keyword to control this needed?

Quote from @mwtoews (https://github.com/shapely/shapely/pull/1497#issuecomment-1243513905):

> One question is with the `add_points` option. The defaults are `add_points=True`, but there are 89 instances in the code with `add_points=False`. Should this keyword be removed? As demonstrated in the code, if a user wants to add points to a line or polygon, they can simply add `plot_points(a, ax=ax, ...)`.

With the merged PR / main branch, we currently get the following behaviour:

```
import shapely
from shapely.plotting import plot_polygon

line = shapely.LinearRing([[0, 0], [50, (100**2-50**2)**0.5], [100, 0]])
polygon = line.buffer(10, 2)

plot_polygon(polygon)
```

<img src="https://user-images.githubusercontent.com/1020496/192099503-7bd8e18e-d63f-4db7-8dd6-33d644c12163.png" width="500">

So the default behaviour is to also add points for the vertices. Currently, this can be turned of using `add_points=False`:

```
plot_polygon(polygon, add_points=False)
```

<img src="https://user-images.githubusercontent.com/1020496/192099649-31b014b1-3af4-47d8-8285-d16099853e34.png" width="500">

If we would change the default of `add_points` to False, or in that case just remove the keyword altogether, users can still do:

```
plot_polygon(polygon)
plot_points(polygon)
# or if the keyword would be kept
plot_polygon(polygon, add_points=True)
```

**Comment by akadouri** [comment]
I think of showing the points as a "debug" feature, which I would often use if I'm working on with a method that manipulates the polygon and trying to see what is happening with the underlying geometry (ex `remove_repeated_points` at different tolerances). That doesn't mean I can't use both methods though, would just be convenient to use one.

**Comment by mwtoews** [comment]
My preference is still with a default `add_points=False`, because it is closer to the "default" style that you'd see with (e.g.) QGIS, where you see line and polygon layers showing simple styles with a solid line.

Also, many "real world" geometries contain thousands of coordinates, and would look strange along the edges, which would consist of many layers of stacked points, e.g.
![image](https://user-images.githubusercontent.com/895458/192131453-0d36cf41-7f12-48ac-b0e8-92fdbb20426a.png)
![image](https://user-images.githubusercontent.com/895458/192131460-f196df84-c492-4d5e-9000-4b4432a95bfe.png)
![image](https://user-images.githubusercontent.com/895458/192131470-6178192e-925c-4bba-a08c-34da008e5120.png)
![image](https://user-images.githubusercontent.com/895458/192131477-0dad2ddc-db20-4be7-86d5-163b92b57038.png)

**Comment by jorisvandenbossche** [comment]
@mwtoews yep, that's a good point ;) For larger geometries with many vertices, adding those points is indeed not a good default. 

It might still boil down to how you will typically use this functionality. For example for me personally, I will typically use it in a "debug" context (eg I used it yesterday to get a visual idea of the toy geometries in https://github.com/shapely/shapely/issues/1551), where you often have either small dummy geometries, or if having real-world geometries you might zoom in to a small area to see what is going on. 
While if I really want to just plot real-world geometries for visualization purposes, I will rather use geopandas' plotting functionality (but that's of course because I am personally already using geopandas).

Just throwing out a possible idea that we could set the default to `add_points=None` and have some threshold number to determine whether the user is plotting a "small" vs "large" geometry, and then add or don't add points by default. 
(although I am not yet convinced myself whether I would actually like such input-dependent behaviour ..; but it would give a nicer default for the extreme (clear) cases. But I can certainly live with a default of `add_points=False` as well)

**Comment by mwtoews** [comment]
A flip-side of my previous argument is [JTS TestBuilder](https://www.eclipse.org/community/eclipse_newsletter/2018/december/jts.php), which I think is really useful to inspect geometries and operations, and adds small square vertices by default. The full set of default controls look like this:
![Screenshot from 2022-09-29 20-07-41](https://user-images.githubusercontent.com/895458/192963618-4ef806be-062e-4bef-9f24-ff4f5d91191d.png)
Some of these controls might inspire other options for the plot utility, such as orientation and labels.
